app/user/service.py

```diff
@@ -0,0 +1,38 @@
+import abc
+
+from app.errors.excpetion import UnAuthorizedException, NotFoundException
+from app.user.models import UserLoginResponse
+from app.user.repository import BaseUserRepository
+from app.depends.auth import get_token
+
+from cryptography.fernet import Fernet
+
+secret_key = Fernet.generate_key()
+cipher_suite = Fernet(secret_key)
+
+class BaseUserService(abc.ABC):
+    def __init__(self, repository: BaseUserRepository) -> None:
+        self.user_repository = repository
+
+    @abc.abstractmethod
+    async def get_user(self, id: str, pw: str) -> UserLoginResponse:
+        raise NotImplementedError
+
+
+class UserService(BaseUserService):
+    async def get_user(self, id: str, pw: str) -> UserLoginResponse:
+        user = await self.user_repository.findUser(id)
+        if not user:
+            raise NotFoundException
+
+        if user.pw != pw:
+            raise UnAuthorizedException
+        
+        cipher_token = await get_token(user.user_id)
+
+        return (
+            UserLoginResponse(
+                token=cipher_token,
+                nickname=user.id,
+            )
+        )

```

Comment:
get_token 로직을 아래와 같이 Service class의 private method로 리팩터링하는 건 어떨까요?(p3)
1. get_token은 depends로 주입될때 사용되는 함수가 아니고, 해당 서비스에서만 사용됩니다.
2. get_token은 내부에 await 하는 로직이 아니므로 코루틴(async)으로 선언할 이유가 없습니다.

```suggestion
        cipher_token = self._get_token(user.user_id)

        return UserLoginResponse(
            token=cipher_token,
            nickname=user.id,
        )

    def _get_token(self, id: str):
        token = bytes(f"{id}-Simple-token-{datetime.now()}", "utf-8")
        cipher_token = cipher_suite.encrypt(token)
        return cipher_token


```

Reply:
현재 cipher_suite를 사용하는 함수가 get_current_user(인가 함수) 와 get_token(토큰생성 함수) 입니다.
cipher_suite를 사용하기 위해서는 secret_key 를 생성해야하는데 만약 auth.py와 user/service.py에서 생성을 하게되면 해당 값이 달라져 문제가 발생합니다. 하여, 위와 같이 auth.py에 생성 후 import 하여 사용했습니다.